Update readme

torkjels opened the pull request:
Mainly, I added a section on Fundraisingbox to the readme.
I also did some small polishing of some sentences, and rearranged some sections.

sarunast commented:
Deploy preview for *masifunde* ready!

Built with commit fcc3765a03846a5a2065297265f5ca466368301c

https://deploy-preview-147--masifunde.netlify.com

erikjalevik commented on README.md:
```diff
@@ -33,33 +34,65 @@ To generate:
+4. Paste your ngrok address into the field "Einbettungsadresse".
+5. Save.
+6. If you go to your ngrok address you should now be able to see the Fundraising form.
+
```
Great stuff about FRB already, but would it help to add info about which form name corresponds to which form on the website?

torkjels replied:
Yeah we can do that. Right now they're called "Development - spenden" and "Development - pate werden", but I guess I could just call them "Spenden" and "Pate werden" in the readme so we dont have to go back and change it later (which we probably will forget to do)

erikjalevik commented on README.md:
```diff
@@ -91,14 +124,13 @@ For each section of the page, we prefix the fields with "section<number>":
-Log into Masifunde's Fundraisingbox (https://secure.fundraisingbox.com). In the top right hover "Einstellung" and click "Spendenformular". In the "Welches Formular wollen Sie bearbeiten" dropdown select the correct Fundraising form (as of writing this, that's "Development"). Paste your ngrok address into the field "Einbettungsadresse". Save. If you go to your ngrok address you should now be able  ...
+Project uses [ESLint AirBnb style guide](https://github.com/airbnb/javascript) with some tweaks (like no semicolons). The `precommit` git hook will
+automatically run ESLint to check if the code complies with the rules. If it fails then it will not
+push to the repo.
```
Maybe add a section on usage of Bootstrap grid together with Styled Components? Like the reason why we have both styled components and the direct class names on elements?

torkjels replied:
Yeah, could mention what we use Bootstrap for, and that it's mostly for the responsive grid+input fields look

erikjalevik commented on README.md:
```diff
@@ -33,33 +34,65 @@ To generate:
+The site is hosted on [Netlify](https://www.netlify.com/), which requires static files.
+Netlify will automatically run `npm run build-static` whenever anything is pushed to the `master` branch since it is connected to this GitHub repository. After that the changes should be reachable on https://masifunde.netlify.com.
+
+When creating a Pull Request (PR), to the branches `master` or `release-1.0`, Netlify will build and publish the merge result as a preview under `deploy-preview-[preview-number]--masifunde.netlify.com`. Netlify can be configured to build deploy previews for other branches as well.
```
Should stuff about the two different branches be removed now?

torkjels commented:
@erikjalevik I've removed the mention of the release-1.0 branch and added a tiny section about Boostrap + styled-components, although I didn't feel like there was much to say. But explaining why we use Bootstrap and what we use it for is good to have.

erikjalevik commented on README.md:
```diff
@@ -50,6 +50,13 @@ When creating a Pull Request (PR) to `master`, Netlify will build and publish th
 * Routes are defined in `routes.js` using [next-routes](https://www.npmjs.com/package/next-routes).
 
+## Bootstrap + styled-components
+We use Bootstrap 4 for two puroses.
```
-> purposes

torkjels commented:
@erikjalevik Fixed the typo 👍

denisw commented on README.md:
```diff
@@ -83,13 +80,56 @@ setting. In that case, try removing the `.next` cache folder.
-Project uses ESLint AirBnb style guide without semicolons. The `precommit` git hook will
-automatically run ESLint to check if the code complies with the rules. If it fails then it will not
-push to the repo.
+Our components are given custom styling using [styled-components](https://www.styled-components.com/) - a css-in-js library.
```
Nitpicking, but "CSS-in-JS"

denisw commented on README.md:
```diff
@@ -83,13 +80,56 @@ setting. In that case, try removing the `.next` cache folder.
+To fully customize the form, we hide most of the inputs in iframe, while rendering our own form fields outside of the iframe. We then [populate the iframe with these custom form field values](https://developer.fundraisingbox.com/v1.0/docs/form-prepopulation-api). This forces a reload and rerender of the iframe, which causes flickering. So we only do this when all custom form fields have values.
+
+### Einbettungsadresse (form embed url)
+The `Einbettungsadresse` should specify the full url of the page this form is used (e.g `https://www.masifunde.de/wie-sie-helfen/spenden`), and not just the host `https://www.masifunde.de`. 
```
For terms like _Einbettungsadresse_ that are not code identifiers, it's better to use _emphasis_ or "quotes".

denisw commented on README.md:
```diff
@@ -121,14 +161,13 @@ For each section of the page, we prefix the fields with "section<number>":
+## Tests
 
-## Testing Fundraisingbox locally
+No tests for now.
```
Expecting a sad emoji here.

denisw commented on README.md:
```diff
@@ -121,14 +161,13 @@ For each section of the page, we prefix the fields with "section<number>":
+## Style guide
 
-Log into Masifunde's Fundraisingbox (https://secure.fundraisingbox.com). In the top right hover "Einstellung" and click "Spendenformular". In the "Welches Formular wollen Sie bearbeiten" dropdown select the correct Fundraising form (as of writing this, that's "Development"). Paste your ngrok address into the field "Einbettungsadresse". Save. If you go to your ngrok address you should now be able  ...
+Project uses [ESLint AirBnb style guide](https://github.com/airbnb/javascript) with some tweaks (like no semicolons). The `precommit` git hook will
```
"AirBnB" with a capital B at the end. (Yeah, more nitpicking.)

torkjels replied:
Actually, it's apparently "Airbnb"! So we're both wrong ❤️ 
https://github.com/airbnb/javascript
https://en.wikipedia.org/wiki/Airbnb
